app/frontend/src/features/auth/signup/AccountForm.test.tsx

```diff
@@ -0,0 +1,200 @@
+import userEvent from "@testing-library/user-event";
+import { EditLocationMapProps } from "components/EditLocationMap";
+import {
+  BIRTHDATE_LABEL,
+  BIRTHDAY_PAST_ERROR,
+  GENDER_REQUIRED,
+  REQUIRED,
+  SIGN_UP,
+  SIGN_UP_BIRTHDAY,
+  SIGN_UP_LOCATION_MISSING,
+  SIGN_UP_TOS_ACCEPT,
+  SIGN_UP_USERNAME_ERROR,
+  USERNAME,
+  USERNAME_REQUIRED,
+} from "features/auth/constants";
+import { FEMALE, HOSTING_STATUS } from "features/constants";
+import { hostingStatusLabels } from "features/profile/constants";
+import { HostingStatus } from "proto/api_pb";
+import { service } from "service";
+import wrapper from "test/hookWrapper";
+import { MockedService } from "test/utils";
+
+import AccountForm from "./AccountForm";
+
+const signupFlowAccountMock = service.auth.signupFlowAccount as MockedService<
+  typeof service.auth.signupFlowAccount
+>;
+const validateUsernameMock = service.auth.validateUsername as MockedService<
+  typeof service.auth.validateUsername
+>;
+
+jest.mock("components/EditLocationMap", () => ({
+  __esModule: true,
+  default: (props: EditLocationMapProps) => (
+    <input
+      data-testid="edit-location-map"
+      onChange={(event) => {
+        props.updateLocation({
+          lat: 1,
+          lng: 2,
+          address: event.target.value,
+          radius: 5,
+        });
+      }}
+    />
+  ),
+}));
+
+describe("AccountForm", () => {
+  beforeEach(() => {
+    signupFlowAccountMock.mockResolvedValue({
+      flowToken: "token",
+      needBasic: false,
+      needAccount: false,
+      needFeedback: true,
+      needVerifyEmail: false,
+    });
+    validateUsernameMock.mockResolvedValue(true);
+  });
+
+  describe("from a filled form", () => {
+    beforeEach(async () => {
+      window.localStorage.setItem(
+        "auth.flowState",
+        JSON.stringify({
+          flowToken: "token",
+          needBasic: false,
+          needAccount: true,
+          needFeedback: false,
+          needVerifyEmail: false,
+        })
+      );
+      render(<AccountForm />, { wrapper });
+
+      userEvent.type(await screen.findByLabelText(USERNAME), "test");
+      const birthdayField = screen.getByLabelText(SIGN_UP_BIRTHDAY);
+      userEvent.clear(birthdayField);
+      userEvent.type(birthdayField, "01/01/1990");
+
+      userEvent.type(
+        screen.getByTestId("edit-location-map"),
+        "test city, test country"
+      );
+
+      userEvent.click(screen.getByLabelText(HOSTING_STATUS));
+      const hostingStatusItem = await screen.findByText(
+        hostingStatusLabels[HostingStatus.HOSTING_STATUS_CAN_HOST]
+      );
+      userEvent.click(hostingStatusItem);
+
+      userEvent.click(screen.getByLabelText(FEMALE));
+      userEvent.click(screen.getByLabelText(SIGN_UP_TOS_ACCEPT));
+    });
+
+    it("submits correctly", async () => {
+      userEvent.click(screen.getByRole("button", { name: SIGN_UP }));
+
+      await waitFor(() => {
+        expect(signupFlowAccountMock).toHaveBeenCalledWith(
```

Comment:
The `signupFlowAccount` service function should take an object here really considering the number of arguments it has! At least then this would become `toHaveBeenCalledWith({ flowToken: 'token', ... })` etc where we are forced to be explicit, which would make the meaning of each parameter clearer for those that aren't self-explanatory. E.g. what the heck is 1, 2, 5 supposed to mean at the end there without looking up the file? :p